src/transformers/modeling_utils.py

```diff
@@ -1645,6 +1645,12 @@ def can_generate(cls) -> bool:
         # Model class overwrites `generate` (e.g. time series models) -> can generate
         if str(cls.__name__) in str(cls.generate):
             return True
+        # The class inherits from a class that can generate (recursive check) -> can generate
+        for base in cls.__bases__:
+            if not hasattr(base, "can_generate"):
```

Comment:
is this only grandfather check?

Reply:
it's recursive! (it calls `can_generate` of the parent, which in turn may call the granparent's, and so on :D )